03_Modules/Configuration/src/module/api.ts

```diff
@@ -310,6 +325,125 @@ export class ConfigurationModuleApi
+  )
+  async updatePassword(
+      request: FastifyRequest<{
+        Body: UpdateChargingStationPasswordRequest;
+        Querystring: UpdateChargingStationPasswordQueryString;
+      }>,
+  ): Promise<IMessageConfirmation> {
+    const stationId = request.body.stationId;
+    this._logger.debug(`Updating password for ${stationId} station`);
+    if (request.body.password && !isValidPassword(request.body.password)) {
+      return {success: false, payload: 'Invalid password'};
+    }
+    const password = request.body.password || generatePassword();
+
+    if (!request.body.setOnCharger) {
+      try {
+        await this.updatePasswordOnStation(password, stationId, request.query.callbackUrl);
+      } catch (error) {
+        this._logger.warn(`Failed updating password on ${stationId} station`, error);
+        return {success: false, payload: `Failed updating password on ${stationId} station`};
+      }
+    }
+    const variableAttributes = await this.updatePasswordForStation(password, stationId);
+    this._logger.debug(`Successfully updated password for ${stationId} station`);
+    return {success: true, payload: `Updated ${variableAttributes.length} attributes`};
+  }
+
+  private async updatePasswordOnStation(password: string, stationId: string, callbackUrl?: string): Promise<void> {
+    const correlationId = uuidv4();
+    const cacheCallbackPromise: Promise<string | null> =
+        this._module.cache.onChange(
+            correlationId,
+            this._module.config.maxCachingSeconds,
+            stationId,
+        );
+
+    const messageConfirmation = await this._module.sendCall(
+        stationId,
+        'T01', // TODO: adjust when multi-tenancy is implemented
+        CallAction.SetVariables,
+        {
+          setVariableData: [
+            {
+              variable: {name: 'BasicAuthPassword'},
+              attributeValue: password,
+              attributeType: AttributeEnumType.Actual,
+              component: {name: 'SecurityCtrlr'},
+            } as SetVariableDataType
+          ]
+        } as SetVariablesRequest,
+        callbackUrl,
+        correlationId
+    );
+    if (!messageConfirmation.success) {
+      throw new Error(`Failed sending request to ${stationId} station for updating password`);
+    }
+
+    const responseJsonString = await cacheCallbackPromise;
+    if (!responseJsonString) {
+      throw new Error(`${stationId} station did not respond in time for updating password`);
+    }
+
+    const setVariablesResponse: SetVariablesResponse = JSON.parse(responseJsonString);
+    const passwordUpdated = setVariablesResponse.setVariableResult
+        .every(result => result.attributeStatus == SetVariableStatusEnumType.Accepted);
+    if (!passwordUpdated) {
+      throw new Error(`Failure updating password on ${stationId} station`);
+    }
+  }
+
+  private async updatePasswordForStation(password: string, stationId: string): Promise<VariableAttribute[]> {
```

Comment:
Should I move this logic to a service layer instead?

Reply:
There is no such concept that's been universally applied, as of yet. It's fine here for now.